src/sentry/integrations/slack/endpoints/base.py

```diff
@@ -68,10 +76,18 @@ def link_user(self, slack_request: SlackRequest) -> Any:
         )
         return self.reply(slack_request, LINK_USER_MESSAGE.format(associate_url=associate_url))
 
-    def unlink_user(self, slack_request: SlackRequest) -> Any:
+    def unlink_user(self, slack_request: SlackDMRequest) -> Response:
         if not slack_request.has_identity:
             return self.reply(slack_request, NOT_LINKED_MESSAGE)
 
+        if not (
+            slack_request.integration
+            and slack_request.user_id
+            and slack_request.channel_id
+            and slack_request.response_url
+        ):
```

Comment:
optional: Should we add this to the `slack_request` object as a function? If it's never reused, that probably isn't necessary but you could make a `slack_request.is_valid()` that would check if all those fields have values (and anymore we add in the future)

Reply:
This is a really good idea this would solve the problem I'm having getting a `obj.get("field")` to be a `str` instead of an `Optional[str]`.

Reply:
Ok I tried really, really hard but I can't get this check into `validate()`. We'll need to overhaul and refactor `SlackResponse` to deserialize and validate if we want this to be nice but I give up for now.